extensions/nouns/src/nouns-traits.tsx

```diff
@@ -0,0 +1,204 @@
+        />
+      ))}
+    </Grid>
+  );
+}
+
+function NounDetail({ noun_id, background, body, accessory, head, glasses }: NounStats) {
+  const backgroundLabel = traits.background[background] ? traits.background[background].label : "Unknown";
+  const bodyLabel = traits.body[body] ? traits.body[body].label : "Unknown";
+  const accessoryLabel = traits.accessory[accessory] ? traits.accessory[accessory].label : "Unknown";
+  const headLabel = traits.head[head] ? traits.head[head].label : "Unknown";
+  const glassesLabel = traits.glasses[glasses] ? traits.glasses[glasses].label : "Unknown";
+
+  return (
+    <Detail
+      actions={
+        <ActionPanel>
+          <Action.CopyToClipboard title="Copy Noun ID" content={noun_id} />
+          <Action.OpenInBrowser
+            title="View on OpenSea"
+            url={`https://opensea.io/assets/ethereum/0x9c8ff314c9bc7f6e59a9d9225fb22946427edc03/${noun_id}`}
+          />
+          <Action.OpenInBrowser title="Open image as PNG" url={`https://noun.pics/${noun_id}`} />
+          <Action.OpenInBrowser title="Open image as SVG" url={`https://noun.pics/${noun_id}.svg`} />
+        </ActionPanel>
+      }
+      navigationTitle={`Detail for Noun #${noun_id}`}
+      markdown={`<img src="https://noun.pics/${noun_id}"/>`}
+      metadata={
+        <Detail.Metadata>
+          <Detail.Metadata.Label title="Noun ID" text={`#${noun_id}`} />
+          <Detail.Metadata.Label title="Background" text={`${backgroundLabel} (${background})`} />
+          <Detail.Metadata.Label title="Body" text={`${bodyLabel} (${body})`} />
+          <Detail.Metadata.Label title="Accessory" text={`${accessoryLabel} (${accessory})`} />
+          <Detail.Metadata.Label title="Head" text={`${headLabel} (${head})`} />
+          <Detail.Metadata.Label title="Glasses" text={`${glassesLabel} (${glasses})`} />
+          <Detail.Metadata.Separator />
+          <Detail.Metadata.Link
+            title="View on..."
+            text="OpenSea"
+            target={`https://opensea.io/assets/ethereum/0x9c8ff314c9bc7f6e59a9d9225fb22946427edc03/${noun_id}`}
+          />
+          <Detail.Metadata.Link title="Open image..." text="as PNG" target={`https://noun.pics/${noun_id}`} />
+          <Detail.Metadata.Link title="Open image..." text="as SVG" target={`https://noun.pics/${noun_id}.svg`} />
+        </Detail.Metadata>
+      }
+    />
+  );
+}
+
+const getAllTraitLabels = ({ background, body, accessory, head, glasses }: NounStats) => {
+  const keywords = [
+    traits.background[background] ? "background: " + traits.background[background].label : "",
+    traits.body[body] ? "body: " + traits.body[body].label : "",
+    traits.accessory[accessory] ? "accessory: " + traits.accessory[accessory].label : "",
+    traits.head[head] ? "head: " + traits.head[head].label : "",
+    traits.glasses[glasses] ? "glasses: " + traits.glasses[glasses].label : "",
+  ].join(" ");
+
+  return [keywords];
```

Comment:
I've updated this function to return an array with a single string, but ideally it should return an array of strings. this is because the fuzzy search doesnt seemt be working as expected — or at least how i expected it

I think it's still not 100%, depending on the order of the words in the search field, it will either show no results, or some results

@thomaspaulmann